File: YazilimGelistiriciDotNetHomeWork/source/repos/KampHomeWorkJune/Day5Homework5GamerBackEnd/Abstracts/BasePromotionManager.cs
using Day5Homework5GamerBackEnd.Entities;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Text;

namespace Day5Homework5GamerBackEnd.Abstracts
{
    public abstract class BasePromotionManager : IPromotionService
    {
        private double PriceAfterDiscountPromotion;
        private double SaleCouponAmountPromotion;
        public void GetPriceAfterDiscountPromotion(Game game, Promotion promotion)
        {
            PriceAfterDiscountPromotion = game.MarketValue-(game.MarketValue * promotion.DiscountAmountForCurrentPurchase/100);
            Console.WriteLine(game.GameTitle + " oyununun guncel piyasa degeri : " + game.MarketValue );
            Console.WriteLine(promotion.PromotionName + " kampanyasinda Kampanyali fiyati : " +PriceAfterDiscountPromotion);
        }
        public void GetSaleCouponAmountPromotion(Game game, Promotion promotion)
        {
            SaleCouponAmountPromotion = game.GiftAmount + promotion.SaleCouponAmountForNextPurchase;
            Console.WriteLine(promotion.PromotionName + " devam ediyor iki kampanyayi birlestirdik : " + game.GameTitle + " oyununu simdi al " + SaleCouponAmountPromotion +" tutarinda hediye ceki kazan.\n");
        }
        public void AddPromotion(Game game, Promotion promotion)
        {
            Console.WriteLine(promotion.PromotionName + " devam ediyor ");
            double kampanyaIndirim = game.DiscountAmount + promotion.DiscountAmountForCurrentPurchase;
            double kampanyaliFiyat = game.MarketValue - (game.MarketValue * kampanyaIndirim / 100);
            Console.WriteLine("Indirim iki kampanyayi birlestirdik : " + game.GameTitle + " oyunu yuzde " + kampanyaIndirim + " indirimle simdi sadece " + kampanyaliFiyat + "\n");
        }

        public virtual void DeletePromotion(Game game, Promotion promotion)
        {
            Console.WriteLine(promotion.PromotionName + " kampanyamiz sona erdi");
            Console.WriteLine("Bizi takip edin.");

        }

        public void InProcessPromotion(Game game, Promotion promotion)
        {
            DateTime ExpireDate = promotion.FinishDate;
            DateTime LocalDate = DateTime.Now;
            int TimeToExpireMonths = ExpireDate.Month - LocalDate.Month;
            int TimeToExpireDays = ExpireDate.Day - LocalDate.Day;
            Console.WriteLine("Kampanya tarihleri : " + promotion.StartDate + " " + promotion.FinishDate);
            Console.WriteLine("Kampanya bitisine :  " + TimeToExpireMonths + " ay ve " + TimeToExpireDays + " gun var\n");
        }

        public void UpdatePromotion(Game game, Promotion promotion)
        {
            promotion.SaleCouponAmountForNextPurchase += 5;
            double lastChance = game.DiscountAmount + promotion.DiscountAmountForCurrentPurchase +5.0;
            double lastPrice = game.MarketValue - (game.MarketValue * lastChance / 100);
            SaleCouponAmountPromotion = game.GiftAmount + promotion.SaleCouponAmountForNextPurchase;
            Console.WriteLine("Ekstra %5 hediye ceki : " + SaleCouponAmountPromotion);
            Console.WriteLine("Damping Ekstra %5 İndirim! Zararina ! " + game.GameTitle + " : " + game.MarketValue + " fiyatindan yuzde " + lastChance + " indirimle " + lastPrice + " fiyatina geriledi");
        }
    }        
}

File: JavaReactKampHomeWork/homeworkJavaSE1.8June/src/day4Homework3GamerBackEnd/abstracts/BasePromotionManager.java
package day4Homework3GamerBackEnd.abstracts;

import java.util.Calendar;

import day4Homework3GamerBackEnd.entities.Game;
import day4Homework3GamerBackEnd.entities.Promotion;

public abstract class BasePromotionManager implements PromotionService{
	
	@Override
	public void addPromotion(Game game, Promotion promotion) {
		System.out.println(promotion.toString());
		System.out.println(game.toString());
		double totalDiscount= game.getDiscountAmount() + promotion.getDiscountAmountForCurrentPurchase();
		double promotedValue = game.getMarketValue()-(game.getMarketValue()*totalDiscount/100);
		System.out.println("Indirim iki kampanyayi birlestirdik : " + totalDiscount + " indirim ile son fiyat: " + promotedValue);
	}
	@Override
	public void inProcessPromotion(Game game, Promotion promotion) {
		Calendar now2 = promotion.getFinishDate();
		Calendar now = Calendar.getInstance();
		int now3 = now2.get(Calendar.DATE)-now.get(Calendar.DATE);
		int now4 = now2.get(Calendar.MONTH)-now.get(Calendar.MONTH) -1;
		if(now3 <0 || now4 < 0) {
			System.out.println("Kampanya sona erdi");
		}else {
			System.out.println(promotion.getPromotionName() + " kampanyasinin bitisine :  " + now4 + " ay ve " + now3 + " gun var");
		}
	}
	@Override
	public void updatePromotion(Game game, Promotion promotion) {
		double totalDiscount = game.getDiscountAmount() + promotion.getDiscountAmountForCurrentPurchase() + 5;
		double updatedPromotion = game.getMarketValue()- (game.getMarketValue()*totalDiscount/100);
		System.out.println(promotion.getPromotionName() +" kampanyasinda ekstra %5 indirim yuzde " + totalDiscount + " indirimle son fiyat :" + updatedPromotion);
	}
	@Override
	public void deletePromotion(Game game, Promotion promotion) {
		boolean isDeleted =false;
		if(game.getAmountInStock()<=0) {			
			System.out.println(promotion.getPromotionName() + " oyununda Kampanyamiz sona erdi");
			isDeleted=true;
		}
		if(isDeleted) {
			System.out.println("Bizi takip edin.");
		}
	}

}
